js_modules/dagit/packages/core/src/ui/Dialog.tsx

```diff
@@ -0,0 +1,137 @@
+import {Dialog as BlueprintDialog} from '@blueprintjs/core';
+import * as React from 'react';
+import styled, {createGlobalStyle} from 'styled-components/macro';
+
+import {Box} from './Box';
+import {ColorsWIP} from './Colors';
+import {Group} from './Group';
+import {IconName, IconWIP} from './Icon';
+
+interface Props extends Omit<React.ComponentProps<typeof BlueprintDialog>, 'backdropClassName'> {}
+
+export const DialogWIP: React.FC<Props> = (props) => {
+  const {...rest} = props;
+  return <BlueprintDialog {...rest} backdropClassName="dagit-backdrop" className="dagit-dialog" />;
+};
+
+interface HeaderProps {
+  icon?: IconName;
+  label: React.ReactNode;
+}
+
+export const DialogHeader: React.FC<HeaderProps> = (props) => {
+  const {icon, label} = props;
+  return (
+    <Box
+      background={ColorsWIP.Gray50}
+      padding={{vertical: 16, horizontal: 20}}
+      border={{side: 'bottom', width: 1, color: ColorsWIP.Gray200}}
+    >
+      <Group direction="row" spacing={8} alignItems="center">
+        {icon ? <IconWIP name={icon} color={ColorsWIP.Gray800} /> : null}
+        <DialogHeaderText>{label}</DialogHeaderText>
+      </Group>
+    </Box>
+  );
+};
+
+export const DialogBody: React.FC = (props) => {
+  return (
+    <Box padding={{vertical: 16, horizontal: 20}} background={ColorsWIP.White}>
+      {props.children}
+    </Box>
+  );
+};
+
+interface DialogFooterProps {
+  right?: React.ReactNode;
+  left?: React.ReactNode;
+}
+
+export const DialogFooter: React.FC<DialogFooterProps> = (props) => {
+  return (
+    <Box
+      padding={{bottom: 20, top: 8, horizontal: 20}}
+      background={ColorsWIP.White}
+      flex={{direction: 'row', alignItems: 'center', justifyContent: 'space-between'}}
+    >
+      <div>{props.left}</div>
+      <div>{props.right}</div>
+    </Box>
+  );
+};
+
+export const DialogButtons: React.FC = (props) => {
```

Comment:
Could we encourage people to pass fragments as `left / right` above and replace the container divs with these spaced Groups? Seems like people will forget to use this part of the API because it's not the same structure as Blueprint

Reply:
Blueprint has the "actions" class for the button container, so I was trying to mirror that. But I just changed it to be a `buttons: React.ReactNode[]`, whose usage will hopefully be pretty clear.